shard/committee/assignment.go

```diff
@@ -131,47 +140,156 @@ func with400Stakers(
 		stakers[i] = validator
 	}
 
+	for i := range stakers {
+		staker := stakers[i]
+		stakers[i].Stake = new(big.Int).Div(
+			staker.Stake, big.NewInt(int64(len(staker.SlotPubKeys))),
+		)
+	}
+
+	unsortedStakes := make([]int, len(stakers))
+	eposStakes := make([]*big.Int, len(stakers))
+
+	for i, j := range stakers {
+		unsortedStakes[i] = int(j.Stake.Int64())
+		eposStakes[i] = j.Stake
+	}
+
+	s3 := effective.Apply(eposStakes)
+
 	sort.SliceStable(
 		stakers,
 		func(i, j int) bool { return stakers[i].Stake.Cmp(stakers[j].Stake) >= 0 },
 	)
-	const sCount = 401
-	top := stakers[:sCount]
+
+	// for i, j := range stakers {
+	// 	sortedStakes[i] = int(j.Stake.Int64())
+	// }
+
+	type t struct {
+		Stakes []int
+	}
+
+	t2 := t{make([]int, len(eposStakes))}
+	for i := range s3 {
+		t2.Stakes[i] = int(s3[i].TruncateInt64())
+	}
+
+	s1, _ := json.Marshal(t{unsortedStakes})
+	s2, _ := json.Marshal(t2)
+
+	fmt.Println("Unsorted")
+	fmt.Println(string(s1))
+	fmt.Println("as EPOS")
+	fmt.Println(string(s2))
+
+	// fmt.Println("Sorted stakers %+v\n", stakers)
+
 	shardCount := int(s.NumShards())
 	superComm := make(shard.State, shardCount)
 	fillCount := make([]int, shardCount)
-	// TODO Finish this logic, not correct, need to operate EPoS on slot level,
-	// not validator level
 
 	for i := 0; i < shardCount; i++ {
-		superComm[i] = shard.Committee{}
-		superComm[i].NodeList = make(shard.NodeIDList, s.NumNodesPerShard())
+		superComm[i] = shard.Committee{
+			uint32(i), make(shard.NodeIDList, s.NumNodesPerShard()),
+		}
 	}
 
-	scratchPad := &bls.PublicKey{}
-
-	for i := range top {
-		spot := int(top[i].Address.Big().Int64()) % shardCount
-		fillCount[spot]++
-		// scratchPad.DeserializeHexStr()
-		pubKey := shard.BlsPublicKey{}
-		pubKey.FromLibBLSPublicKey(scratchPad)
-		superComm[spot].NodeList = append(
-			superComm[spot].NodeList,
-			shard.NodeID{
-				top[i].Address,
+	shardBig := big.NewInt(int64(s.NumShards()))
+
+	for i := 0; i < len(s.FnAccounts()); i++ {
```

Comment:
This should be hardcoded 320. Not len(s.FnAccounts()). FnAccounts will be deprecated after staking.